src/components/Search/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { PureComponent } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'proptypes';
+import './styles.scss';
+
+/* eslint-disable no-restricted-globals */
+class Search extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      searchOptionsVisible: false,
+      searchValue: '',
+      searchCriteria: '',
+    };
+  }
+
+  componentDidUpdate() {
+    const { searchValue, searchCriteria } = this.state;
+    const { handleSearch } = this.props;
+    if (searchValue || searchCriteria) {
+      handleSearch(searchValue, searchCriteria);
+    }
+  }
+
+  toggleVisibility = () => {
+    this.setState(prevState => ({
+      searchOptionsVisible: !prevState.searchOptionsVisible,
+    }));
+  };
+
+  handleSearchCriteriaChange = (optionId) => {
+    this.setState({ searchCriteria: optionId });
+  };
+
+  handleSearchValueChange = (event) => {
+    this.setState({ searchValue: event.target.value });
+  }
+
+  handleClickOutside() {
```

Comment:
This `handleClickOutside` is not used anywhere in this component neither is it passed down to the child components, is there something I am missing?